packages/core-products/src/module/configureProductsModule.ts

```diff
@@ -44,11 +44,12 @@ const buildFindSelector = ({
   includeDrafts = false,
   productIds,
   productSelector,
+  queryString,
   ...query
 }: ProductQuery) => {
   const selector: Query = productSelector ? { ...productSelector, ...query } : query;
 
-  if (productIds) {
+  if (productIds?.length) {
```

Comment:
Nope, please revert this line, if productIds is provided no matter if empty array or not, this has to be applied.